recipes/osgearth/all/conanfile.py

```diff
@@ -22,6 +24,33 @@ class OsgearthConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+
+        # osgearth has support for an imgui integration, but it is honestly a mess:
+        #
+        #   - osgearth v3.2 includes an embedded version of imgui (and imnodes) which is used to build the imgui
+        #     nodekit, but only if GLEW exists as a dependency.
+        #
+        #   - osgearth v3.3 removes the embedded version of imgui (and imnodes), effectively rendering the imgui nodekit
+        #     unusable (that is, unless the user actually copies their own version of imgui into the source directory).
+        #
+        #   - osgearth v3.7.2 yet again includes an embedded version of imgui (and imnodes), and yet again, GLEW must
+        #     exist as a dependency in order to build the imgui nodekit. This time, however, there is an additional
+        #     requirement: the OSGEARTH_BUILD_IMGUI_NODEKIT CMake variable must also be set.
+        #
+        # It should be possible to patch osgearth to ensure consistency between all of our supported versions by doing
+        # the following:
+        #
+        #   - Replace the embedded versions of imgui and imnodes with their corresponding ConanFile.requires dependency.
+        #
+        #   - Ensure that the OSGEARTH_BUILD_IMGUI_NODEKIT CMake variable is respected across every osgearth version.
+        #
+        #   - Remove the explicit dependency on GLEW. (NOTE: This would be ideal, since osgearth only ever depends on
+        #     GLEW to support its embedded version of imgui. However, it probably isn't required that we do this.)
+        #
+        # For now, we simply disable the imgui nodekit.
+        #
+        # "build_imgui_nodekit": [True, False],
+
         "build_procedural_nodekit": [True, False],
         # "build_triton_nodekit": [True, False],
```

Comment:
The following options have historically been commented out in this recipe without explanation:

* `build_triton_nodekit`
* `build_silverlining_nodekit`
* `with_basisu`

(NOTE: `with_glew` also existed as a commented-out option prior to this commit. However, it was removed because a `build_imgui_nodekit` option would be better suited in practice, since `osgearth` only really ever used GLEW to support its embedded versions of `imgui`.)

I have not tried re-enabling support for these options in this PR. However, it would be helpful to understand _why_ these options aren't (or perhaps weren't) supported.

Reply:
Thanks for clarifying! I do have another question, though. Is it better to leave these so-called "hypothetical options" as commented-out code in the recipe, or should we just be removing any and all mentions of them?

Reply:
Let's remove them althogether, and users that might have a need for them will know to ask for the feature itself :)